### ISSUE TITLE
[Bug] Regular expression validator for email needs to ignore casing

### ISSUE THREAD

**Issue by jfversluis** [xamarin/XamarinCommunityToolkit#140]
### Description

The regular expression for email right now will mark a value as invalid when you use uppercase letters in the address. This should not happen. Besides this, lets make sure that we have the correct regex for _all_ that is valid in an email address

### Steps to Reproduce

1. Open sample app
2. enter an email address in the right entry containing an uppercase letter but is otherwise valid
3. Observe how the value goes red while it shouldn't

### Expected Behavior

Value is not marked as invalid (showing red)

### Actual Behavior

Value is shown as invalid (showing red)

### Screenshot

![image](https://user-images.githubusercontent.com/939291/84871398-c378d580-b080-11ea-9299-2ee12ce4192c.png)

### Basic Information

- Version with issue: master
- Last known good version: N/A

**Comment by SunnyMukherjee** [comment]
This bug was fixed in the last few commits of PR #126.  I introduced a new regular expression that allows for capital letters.

Here is a screenshot of it working with uppercase letters.

![image](https://user-images.githubusercontent.com/20747117/84908862-0340bc80-b083-11ea-9fbc-646e395f18bd.png)

Here is how it looks when an invalid email is entered.

![image](https://user-images.githubusercontent.com/20747117/84909039-3420f180-b083-11ea-99d9-0d21320517b3.png)
